py/server/deephaven/time.py

```diff
@@ -877,3 +878,24 @@ def to_np_timedelta64(dt: Union[None, Duration, Period]) -> Optional[numpy.timed
         raise DHError(e) from e
 
 # endregion
+
+# region Utility
+
+def simple_date_format(pattern: str) -> jpy.JType:
+    """ Creates a Java SimpleDateFormat from a date-time format pattern.
```

Comment:
Please expand this documentation to suggest that users never call this directly from filter and column expressions, only when setting up format to use in same.

Reply:
I am interpreting Ryan's comment to mean:

"Tell them to not do this:
```
from deephaven.time import simple_date_format
from deephaven import empty_table

t = empty_table(1).update([
    "UnformattedTimestamp = `19440101`",
    "FormattedTimestamp = simple_date_format(`YYYYmmdd`).parse(UnformattedTimestamp)"
])
```

Devin's comment implies that the "anti-pattern" also includes:
```
from deephaven.time import simple_date_format
from deephaven import empty_table

sdf = simple_date_format("YYYYmmdd")
t = empty_table(1).update([
    "UnformattedTimestamp = `19440101`",
    "FormattedTimestamp = sdf.parse(UnformattedTimestamp)"
])
```
Which reduces the scope of "correct" usage to basically zero. Can we clarify the correct usage for this method? My guess would have been the latter is what we want, as it follows convention from `deephaven.time`, and the former is bad.

Reply:
As far as I'm concerned, the second example in Alex's message preceding this one is the one and only use case.

Reply:
My assumption that JType caused boundary crossing was invalid, so I don't have any strict objections.

I think it puts us in a place where we need to be careful in the future - ie, we _can't_ have `simple_date_format` return a `JObjectWrapper` type b/c we may be relying on the fact that it doesn't cross the boundary in a query formula. @chipkent , I think it's worthwhile to have a developer comment in the code remarking on this.